pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java

```diff
@@ -103,6 +103,39 @@
      */
     List<String> getList(String namespace, TopicDomain topicDomain) throws PulsarAdminException;
 
+    /**
+     * Get the list of topics under a namespace.
+     * <p/>
+     * Response example:
+     *
+     * <pre>
+     * <code>["topic://my-tenant/my-namespace/topic-1",
+     *  "topic://my-tenant/my-namespace/topic-2"]</code>
+     * </pre>
+     *
+     * @param namespace
+     *            Namespace name
+     *
+     * @param topicDomain
+     *            use {@link TopicDomain#persistent} to get persistent topics
+     *            use {@link TopicDomain#non_persistent} to get non-persistent topics
+     *            Use null to get both persistent and non-persistent topics
+     *
+     * @param params
+     *            params to filter the results
+     *
+     * @return a list of topics
+     *
+     * @throws NotAuthorizedException
+     *             Don't have admin permission
+     * @throws NotFoundException
+     *             Namespace does not exist
+     * @throws PulsarAdminException
+     *             Unexpected error
+     */
+    List<String> getList(String namespace, TopicDomain topicDomain, Map<String, Object> params)
```

Comment:
IMHO, It's a little confusing to add `params` here. It's easy for coding, but not so easy for user to understand how to use this.

Reply:
yes, added an enum to use query param for usability.